asmcomp/selectgen.ml

```diff
@@ -641,7 +641,7 @@ method emit_expr (env:environment) exp =
       let r = self#regs_for typ_float in
       Some(self#insert_op (Iconst_float (Int64.bits_of_float n)) [||] r)
   | Cconst_symbol n ->
-      let r = self#regs_for typ_val in
+      let r = self#regs_for typ_int in
```

Comment:
The reason why this works is quite intricate, this certainly need some comment to explain it (if you don't end up adding a new register type to represent that).

Reply:
FYI: I had a crash with 4.08 and no-naked-pointers that we narrowed down to another instance of this bug with the help of @mshinwell